source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,105 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = uint;
+
+    ///
+    const(T)[] probs;
+    size_t N;
+    double norm = 0.0;
+
+
+    /++
+    Params:
+        probs = probabilities of the multinomial distribution
+        N = Number of rolls
+    Constraints: `sum(probs[i]) <= 1`
+    +/
+    this()(size_t N, const(T)[] probs)
+    {
+        this.N = N;
+        this.probs = probs;
+         /// Makes sure probabilities add up to one, by calculating a normalization factor
+        version(assert)
+        {
+            T norm = 0;
+            foreach(k, p; this.probs)
+        {
+            norm += p;
+        }
+            assert(fabs(norm - 1) <= T.epsilon * probs.length * 2);
+        }
+
+    }
+
+    ///
+    pragma(inline, false)
+    void opCall(G)(scope ref G gen, scope size_t[] result)
```

Comment:
as was mentioned before it should be `scope uint[] result` as a user will never use huge arrays for ndvariables.

Reply:
If I replace this with uint I get this error:
```D
source/mir/random/ndvariable.d(378,7): Error: template mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall cannot deduce function from argument types !()(MersenneTwisterEngine!(ulong, 64LU, 312LU, 156LU, 31LU, 13043109905998158313LU, 29LU, 6148914691236517205LU, 17LU, 8202884508482404352LU, 37LU, 18444473444759240704LU, 43LU, 6364136223846793005LU), ulong[]), candidates are:
source/mir/random/ndvariable.d(327,10):        mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall(G)(ref scope G gen, scope uint[] result) if (isSaturatedRandomEngine!G)
source/mir/random/ndvariable.d(353,10):        mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall(G)(scope G* gen, scope uint[] result) if (isSaturatedRandomEngine!G)
/usr/bin/dmd failed with exit code 1.
```

Reply:
I recall this issue with the size_t and uint from before. Would something like below be useful:

```
import std.traits : isUnsigned;

template smaller_unsigned(T)
    if (isUnsigned!T)
{
    import std.meta;
    alias unsigned_types = AliasSeq!(ubyte, ushort, uint, ulong);
    static if (is(T == Largest!(unsigned_types[3], T))) {
        alias smaller_unsigned = unsigned_types[2];
    } else static if (is(T == Largest!(unsigned_types[2], T))) {
        alias smaller_unsigned = unsigned_types[1];
    } else static if (is(T == Largest!(unsigned_types[1], T))) {
        alias smaller_unsigned = unsigned_types[0];
    }
}
```

Reply:
simple `uint[]` is good for API simplicity. We just need to update the unittests.

Reply:
done.